pkg/providers/instance/instance.go

```diff
@@ -102,34 +109,26 @@ func (p *DefaultProvider) Create(ctx context.Context, nodeClass *v1alpha1.ECSNod
 }
 
 func (p *DefaultProvider) Get(ctx context.Context, id string) (*Instance, error) {
-	describeInstancesRequest := &ecsclient.DescribeInstancesRequest{
-		RegionId:    tea.String(p.region),
-		InstanceIds: tea.String("[\"" + id + "\"]"),
+	if instance, ok := p.instanceCache.Get(id); ok {
+		return instance.(*Instance), nil
 	}
-	runtime := &util.RuntimeOptions{}
 
-	resp, err := p.ecsClient.DescribeInstancesWithOptions(describeInstancesRequest, runtime)
+	// List all instances to update the cache
+	instances, err := p.List(ctx)
```

Comment:
Why don't we only cache the instance with specific id here

Reply:
If there is multiple instances, use list to retrive all  in the first request and the next will go through the cache.

If with specific id, eatch one will do a request.